text/0000-pack-subcommands.md

```diff
@@ -0,0 +1,275 @@
+
+* add
+* remove
+* list
+* (implied list)
+
+The explicit additional parameters would be based on the specific implementation needs.
+
+###### Example
+
+```
+# adds `gcr.io/my-org/my-image` as a mirror for `my-image`  
+pack config run-image-mirrors add my-image gcr.io/my-org/my-image
+
+# removes `gcr.io/my-org/my-image` as a mirror for `my-image`
+pack config run-image-mirrors remove my-image gcr.io/my-org/my-image
+
+# removes all mirrors for `my-image`
+pack config run-image-mirrors remove my-image
+
+# list all mirrors
+pack config run-image-mirrors list
+
+# list all mirrors (implied list)
+pack config run-image-mirrors
+
+# list all mirrors for `my-image`
+pack config run-image-mirrors list my-image
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+
+## Backwards Compatibility
+
+1. There would be a period in which all pre-existing subcommands would work (with a deprecation warning).
+2. Critical operations (`build`, `rebase`) will remain unchanged.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Change is bad... mkay.
+
+# Alternatives
+[alternatives]: #alternatives
+
+### Colon separated commands
+
+> We could use a `pack <topic>:<command>` pattern instead of space separated.
+
+Some examples:
+
+**Single config resource:**
+
+```shell
+# set
+pack config:default-builder gcr.io/my-org/my-builder:latest
+
+# unset
+pack config:default-builder:unset
+
+# get
+pack config:default-builder:get
+
+# get (implied)
+pack config:default-builder
+```
+
+**List config resource:**
+
+```shell
+# adds `gcr.io/my-org/my-image` as a mirror for `my-image`  
+pack config:run-image-mirrors:add my-image gcr.io/my-org/my-image
+
+# removes `gcr.io/my-org/my-image` as a mirror for `my-image`
+pack config:run-image-mirrors:remove my-image gcr.io/my-org/my-image
+
+# removes all mirrors for `my-image`
+pack config:run-image-mirrors:remove my-image
+
+# list all mirrors (implied)
+pack config:run-image-mirrors
+
+# list all mirrors
+pack config:run-image-mirrors:list
+
+# list all mirrors for `my-image`
+pack config:run-image-mirrors:list my-image
+```
+
+### Alternative Utility
+
+> Another alternative is to focus "pack" more for specific audiences which would reduce the number of commands that it would have to know.
+
+```
+pack
+
+  build
+  config
+     ...
+  inspect
+  run
+```
+
+The alternative utility CLI would be everything else:
+
+```
+bputils
+
+  builder
+    create
+    inspect
+    suggest
+  buildpack
+    create
+    package
+    register
+  stack
+    suggest
+```
+
+### Config alternatives
+
+- Only allow manual edits of config file.
+    - A `config` command would allow providing `help` for possible operations and argument values.
+- Some [Config](#config) subcommands could be more intertwined with their respective resource. For example, `pack builder trust ...` instead of `pack config trust-builder ...`.
+    - Other commands are operations on the resource whereas having `config` subcommands would make it more obvious that
```

Comment:
I like this point a lot. I think it makes the UX a bit more sensible, though sometimes wordy.